File: 0069-sqrtx/0069-sqrtx.cpp
class Solution {
public:
    int mySqrt(int x) {
        int low=1,high=x;
        while(low<=high){
            double mid=low + (high - low) / 2;;
            if(mid*mid==x){return mid;}
            if(mid*mid<=x){
                low=mid+1;
            }else{high=mid-1;}
        }
        return high;
    }
};

File: 0069-sqrtx/0069-sqrtx.py
class Solution:
    def mySqrt(self, x: int) -> int:
        l = 2
        r  = x//2
        if x<2:
            return x
        while(l<=r):
            mid = l+(r-l)//2
            chk = mid*mid
            if chk ==x:
                return (mid)
            elif chk>x:
                r = mid-1
            else: 
                l = mid+1
            
        return r
        

        